File: app/src/main/java/com/funckyhacker/fileexplorer/di/MainComponent.java
package com.funckyhacker.fileexplorer.di;

import com.funckyhacker.fileexplorer.MainActivity;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface MainComponent extends AndroidInjector<MainActivity>{
  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<MainActivity> {
  }
}

File: app/src/main/java/com/funckyhacker/fileexplorer/di/MainComponent.kt
package com.funckyhacker.fileexplorer.di

import com.funckyhacker.fileexplorer.view.main.MainActivity
import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface MainComponent : AndroidInjector<MainActivity> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<MainActivity>()
}
